Modules/_winapi.c

```diff
@@ -1920,36 +1909,31 @@ static PyMethodDef winapi_functions[] = {
     {NULL, NULL}
 };
 
-static struct PyModuleDef winapi_module = {
-    PyModuleDef_HEAD_INIT,
-    "_winapi",
-    NULL,
-    -1,
-    winapi_functions,
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
-#define WINAPI_CONSTANT(fmt, con) \
-    PyDict_SetItemString(d, #con, Py_BuildValue(fmt, con))
+#define WINAPI_CONSTANT(fmt, con) { \
+    PyObject *value = Py_BuildValue(fmt, con); \
+    if (PyDict_SetItemString(d, #con, value) < 0) { \
+        Py_DECREF(value); \
+        return -1; \
+    } \
+    Py_DECREF(value); \
+}
 
-PyMODINIT_FUNC
-PyInit__winapi(void)
+static int winapi_exec(PyObject *m)
 {
-    PyObject *d;
-    PyObject *m;
+    WinApiState *st = winapi_get_state(m);
 
-    if (PyType_Ready(&OverlappedType) < 0)
-        return NULL;
+    st->overlapped_type = (PyTypeObject *)PyType_FromModuleAndSpec(m, &winapi_overlapped_type_spec, NULL);
+    if (st->overlapped_type == NULL) {
+        return -1;
+    }
 
-    m = PyModule_Create(&winapi_module);
-    if (m == NULL)
-        return NULL;
-    d = PyModule_GetDict(m);
+    Py_INCREF(st->overlapped_type);
```

Comment:
PyModule_AddType() doesn't need the ugly INCREF/DECREF dance.

Reply:
@vstinner Good point.  I just looked at the implementation it does all the work